File: April-18/24-04-18/java/03-Solitary-String.java

import java.io.*;


class TestClass {
    static int check(String str)
    {
        int n = str.length();
        int maxi = -1,diff = 0;
        int N = 27;
        int first[] = new int[N];
        int last[] = new int[N];
        
        for(int i = 0; i < N; i++)
        {
            first[i] = last[i] = -1;
        }
        for(int i = 0; i < n; i++)
        {
            int val = str.charAt(i)- 'a';
            if(first[val] == -1)
            {
                first[val] = i;
            }
            else
            {
                last[val] = i;
            }
        }
        
        for(int i = 0; i < N; i++)
        {
            if(first[i] != -1 && last[i] != -1)
            {
                diff = last[i] - first[i] - 1;
                if(diff > maxi)
                    maxi = diff;
            }
        }
        return maxi;
    }
    public static void main(String args[] ) throws Exception {
        BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(input.readLine());
        while(t-- > 0)
        {
            String str = input.readLine();
            System.out.println(check(str));
            
            
        }
        
    }
}

File: April-18/24-04-18/python/03-Solitary-String.py
def check(string):
    n = len(string)
    maxi = -1
    diff = 0
    N = 27
    first = [-1]*N
    last = [-1]*N
    for i in range(0, n):
        val = ord(string[i]) - ord('a')
        if(first[val] == -1):
            first[val] = i
        else:
            last[val] = i
    for i in range(0, N):
        if(first[i] != -1 and last[i] != -1):
            diff = last[i] - first[i] - 1
            if(diff > maxi):
                maxi = diff
    return maxi

for _ in range(int(input())):
    x = input()
    print(check(x))
